packages/notifi-react-card/lib/hooks/useNotifiSubscribe.ts

```diff
@@ -279,37 +299,38 @@ export const useNotifiSubscribe: () => Readonly<{
           );
         }
 
+        const alert = await client.createAlert({
+          emailAddress: finalEmail,
+          filterId: filter?.id ?? '',
+          filterOptions: filterOptions ?? undefined,
+          groupName: 'managed',
+          name: currentConfigurationData.alertName,
+          phoneNumber: finalPhoneNumber,
+          sourceId: source?.id ?? '',
+          targetGroupName: 'Default',
+          telegramId: finalTelegramId,
+        });
+
+        newResults[currentConfigurationData.alertName] = alert;
+      }
+    } else {
+      const alerts = data.alerts;
+
+      for (let i = 0; i < alerts.length; i++) {
```

Comment:
this part is not actually sufficient

when we have alerts, we need to:
- identify which alertConfigurations have a corresponding alert
  - if the corresponding alert is the same (i.e. same source, same target) do not do anything
  - if the corresponding alert is not the same, delete it and recreate it using configuration
- identify which alerts have a `null` alertConfiguration and delete them
- if there exists an alert whose name doesn't appear in alertConfiguration, leave them alone

happy to talk through what i mean